src/routes/NewUser/components/NewUserPage/NewUserPage.js

```diff
@@ -94,12 +95,67 @@ function NewUser() {
     errors,
     register,
     formState: { isValid, dirty },
+    getValues,
+    setValue,
   } = useForm({
     validationSchema: userProfileSchema,
     defaultValues,
   });
   const [userLocationInfo, setUserLocationInfo] = useState(null);
 
+  /**
+   * Sets the value for the given field if it isn't already populated.
+   * @param {string} fieldName name of the field which will be set
+   * @param {string} allFieldValues current value of all fields
+   * @param {object} googleResultComponent address component returned by the Google Maps API
+   * @param {object} addressType type of address component to lookup
+   * @param {string} addressTypeFormat short_name or long_name
+   */
+  function setValueIfNotAlreadySet(
+    fieldName,
+    allFieldValues,
+    googleResultComponent,
+    addressType,
+    addressTypeFormat = 'long_name',
+  ) {
+    if (
+      googleResultComponent.types.indexOf(addressType) !== -1 &&
+      !allFieldValues[fieldName]
+    ) {
+      setValue(fieldName, googleResultComponent[addressTypeFormat]);
+    }
+  }
+
+  function handleSetUserLocationInfo(locationInfo) {
+    setUserLocationInfo(locationInfo);
+    const result = locationInfo.lookedUpAddress;
+    if (result) {
+      const values = getValues();
+      let streetNumber = '';
+      let streetAddress = '';
+      result.address_components.forEach((component) => {
+        setValueIfNotAlreadySet('city', values, component, 'locality');
+        setValueIfNotAlreadySet(
+          'state',
+          values,
+          component,
+          'administrative_area_level_1',
+          'short_name',
+        );
+        setValueIfNotAlreadySet('zipcode', values, component, 'postal_code');
+        if (component.types.indexOf('street_number') !== -1) {
```

Comment:
Same here, if you are just checking for existinence instead of finding the actual index, we can use includes:

```js
component.types.includes('street_number')
```